tests/runner/src/main/java/org/mule/test/runner/api/AetherClassPathClassifier.java

```diff
@@ -655,6 +666,23 @@ private Artifact createPluginArtifact(String pluginCoords, Artifact rootArtifact
   }
 
   /**
+   * Resolves the rootArtifact {@value #JAR_EXTENSION} output {@link File}s to be added to class loader.
+   *
+   * @param rootArtifact {@link Artifact} being classified
+   * @return {@link File} to be added to class loader
+   */
+  private File resolveRootArtifactFile(Artifact rootArtifact) {
+    final DefaultArtifact jarArtifact = new DefaultArtifact(rootArtifact.getGroupId(), rootArtifact.getArtifactId(),
+                                                            JAR_EXTENSION, JAR_EXTENSION, rootArtifact.getVersion());
+    try {
+      return dependencyResolver.resolveArtifact(jarArtifact).getArtifact().getFile();
+    } catch (ArtifactResolutionException e) {
+      logger.warn("'{}' rootArtifact output {} couldn't be resolved", rootArtifact, JAR_EXTENSION);
+      return null;
```

Comment:
Because a rootAritfact classified as APP could have or not a target/classes. For instance, the case of testing artifacts (it tests or http tests) they are APP artifacts from classification point of view but they don't have a  target/classes.